src/test/scala/com/unstablebuild/slime/TypeEncodersTest.scala

```diff
@@ -0,0 +1,116 @@
+  }
+
+  it must "encode types keyed by symbol or string" in {
+
+    encode('hello -> "world") must equal(Seq("hello" -> StringValue("world")))
+    encode('hello -> 123) must equal(Seq("hello" -> NumberValue(123)))
+    encode('hello -> true) must equal(Seq("hello" -> BooleanValue(true)))
+    encode('hello -> 'y') must equal(Seq("hello" -> StringValue("y")))
+  }
+
+  it must "encode collections" in {
+
+    encode("hello" -> Array(1, 2, 3)) must equal(
+      Seq("hello" -> SeqValue(Seq(NumberValue(1), NumberValue(2), NumberValue(3))))
+    )
+    encode("hello" -> Array('a -> ('b -> 2))) must equal(
+      Seq("hello" -> NestedValue(Seq("a" -> NestedValue(Seq("b" -> NumberValue(2))))))
+    )
+    encode(Array('a -> 1, 'b -> 2)) must equal(Seq("a" -> NumberValue(1), "b" -> NumberValue(2)))
+
+    encode("hello" -> Seq(1, 2, 3)) must equal(
+      Seq("hello" -> SeqValue(Seq(NumberValue(1), NumberValue(2), NumberValue(3))))
+    )
+    encode("hello" -> List(1, 2, 3)) must equal(
+      Seq("hello" -> SeqValue(Seq(NumberValue(1), NumberValue(2), NumberValue(3))))
+    )
+    encode("hello" -> mutable.ListBuffer(1, 2, 3)) must equal(
+      Seq("hello" -> SeqValue(Seq(NumberValue(1), NumberValue(2), NumberValue(3))))
+    )
+
+    encode("hello" -> Set(1)) must equal(Seq("hello" -> SeqValue(Seq(NumberValue(1)))))
+    encode("hello" -> immutable.HashSet(1)) must equal(Seq("hello" -> SeqValue(Seq(NumberValue(1)))))
+
+    encode("hello" -> Set('a -> 1)) must equal(Seq("hello" -> NestedValue(Seq("a" -> NumberValue(1)))))
+    encode("hello" -> Set('a -> ('b -> 2))) must equal(
+      Seq("hello" -> NestedValue(Seq("a" -> NestedValue(Seq("b" -> NumberValue(2))))))
+    )
+
+    encode("hello" -> Seq(1, 2, 3).par) must equal(
+      Seq("hello" -> SeqValue(Seq(NumberValue(1), NumberValue(2), NumberValue(3))))
+    )
+
+    encode(Map('a -> 1, 'b -> 2)) must equal(Seq("a" -> NumberValue(1), "b" -> NumberValue(2)))
+    encode("hello" -> Map('a -> 1, 'b -> 2)) must equal(
+      Seq("hello" -> NestedValue(Seq("a" -> NumberValue(1), "b" -> NumberValue(2))))
+    )
+    encode("hello" -> Map('a -> Map("a" -> 1), 'b -> Map("b" -> 2))) must equal(
+      Seq(
+        "hello" -> NestedValue(
+          Seq("a" -> NestedValue(Seq("a" -> NumberValue(1))), "b" -> NestedValue(Seq("b" -> NumberValue(2))))
+        )
+      )
+    )
+
+    encode(Seq('a -> 1, 'b -> 2)) must equal(Seq("a" -> NumberValue(1), "b" -> NumberValue(2)))
+    encode("hello" -> Seq('a -> 1, 'b -> 2)) must equal(
+      Seq("hello" -> NestedValue(Seq("a" -> NumberValue(1), "b" -> NumberValue(2))))
+    )
+  }
+
+  it must "encode options" in {
+    encode("hello" -> Option.empty[String]) must equal(Seq("hello" -> SeqValue(Seq.empty)))
+    encode("hello" -> Option("hi")) must equal(Seq("hello" -> SeqValue(Seq(StringValue("hi")))))
+  }
+
+  it must "encode encodable types" in {
+
+    case class MyPair(int: Int, str: String) extends Encodable {
```

Comment:
Do you think we will need this OO pattern? Maybe it is less confusing (i.e. to only provide one way of doing things) to only support defining the encoder as an implicit? If someone needs this it is trivial to create an implicit encoder that encodes a self defined type like "Encodable".